homeassistant/components/sensor/zamg.py

```diff
@@ -0,0 +1,255 @@
+            "friendly_name": SENSOR_TYPES[self.variable][0],
+            "station": self.probe.get_data('station_name'),
+            "updated": "%s %s" % (self.probe.get_data('update_date'),
+                                  self.probe.get_data('update_time'))
+
+        }
+
+
+# ------------------------------------------------------------------------
+class ZamgData(object):
+    """
+    I represent weather data for a specific site.
+
+    From the web site:
+
+    Sie beinhalten neben Stationsnummer, Stationsname, Seehöhe der Station,
+    Messdatum und Messzeit (Lokalzeit) die meteorologischen Messwerte von
+    Temperatur, Taupunkt, relative Luftfeuchtigkeit, Richtung und
+    Geschwindigkeit des Windmittels und der Windspitze, Niederschlagssumme
+    der letzten Stunde, Luftdruck reduziert auf Meeresniveau und Luftdruck
+    auf Stationsniveau sowie die Sonnenscheindauer der letzten Stunde (in
+    Prozent). Die Messstationen, die diese Daten liefern, sind über das
+    Bundesgebiet verteilt und beinhalten alle Landeshauptstädte sowie
+    die wichtigsten Bergstationen.
+    """
+
+    API_URL = "http://www.zamg.ac.at/ogd/"
+
+    API_FIELDS = {
+        v[2]: (k, v[3])
+        for k, v in SENSOR_TYPES.items()
+    }
+
+    API_HEADERS = {
+        'User-Agent': 'home-assistant.zamg/' + __version__,
+    }
+
+    def __init__(self, logger, station_id):
+        """Initialize the probe."""
+        self._logger = logger
+        self._station_id = station_id
+        self.data = {}
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """
+        Update data set.
+
+        Fetch a new data set from the zamg server, parse it and
+        update internal state accordingly
+        """
+        response = requests.get(self.API_URL,
+                                headers=self.API_HEADERS, timeout=15)
+        response.encoding = 'UTF8'
+        if response.status_code == 200:
+            content_type = response.headers.get('Content-Type', 'whatever')
+            if content_type == 'text/csv':
+                content = response.text
+                data = (line for line in content.split('\n'))
+                reader = csv.DictReader(data, delimiter=';', quotechar='"')
+                for row in reader:
+                    if row.get("Station", None) == self._station_id:
+                        self.data = {
+                            self.API_FIELDS.get(k)[0]:
+                                self.API_FIELDS.get(k)[1](v.replace(',', '.'))
+                            for k, v in row.items()
+                            if v and k in self.API_FIELDS
+                        }
+                        break
+            else:
+                self._logger.error("Expected text/csv but got %s",
+                                   content_type)
+        else:
+            self._logger.error("API call returned with status %s",
+                               response.status_code)
```

Comment:
I think that's worth to catch `requests.exceptions.RequestException` for the requests.